projects/demo/src/app/snippet/snippet.component.ts

```diff
@@ -7,20 +7,17 @@ import { switchMap } from 'rxjs';
   selector: 'demo-snippet',
   templateUrl: './snippet.component.html',
   styleUrls: ['./snippet.component.scss'],
-  standalone: true,
   encapsulation: ViewEncapsulation.ShadowDom
 })
 export class SnippetComponent implements AfterViewInit {
+  private highlighterService = inject(HighlighterService);
+  private snippetService = inject(SnippetService);
+
   divEl = viewChild.required<ElementRef<HTMLDivElement>>('code');
 
   code = input.required<string>();
   lang = input.required<SupportedLanguages>();
 
-  constructor(
-    private highlighterService: HighlighterService,
-    private snippetService: SnippetService
-  ) {}
-
   ngAfterViewInit(): void {
     this.snippetService
```

Comment:
This could be done in the constructor, and we should use `takeUntilDestroyed()`.